sale_order_line_effective_date/models/sale_order_line.py

```diff
@@ -0,0 +1,23 @@
+# Copyright 2024 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    effective_date = fields.Datetime(
+        compute="_compute_effective_date",
+        store=True,
+        help="Completion date of the first delivery order.",
+    )
+
+    @api.depends("move_ids.date")
+    def _compute_effective_date(self):
+        for line in self:
+            moves = line.move_ids.filtered(
+                lambda x: x.state == "done" and x.location_dest_id.usage == "customer"
+            )
+            dates_list = [date for date in moves.mapped("date") if date]
+            line.effective_date = min(dates_list, default=False)
```

Comment:
Use [:1] to get an empty Recordset if there is no records in the Recordset and .date will give you False

```suggestion
            line.effective_date = line.move_ids.filtered_domain([
                ("state", "=", "done"),
                ("location_dest_id.usage", "=", "customer"),
                ("date", "!=", False),
            ]).sorted("date")[:1].date
```